app/services/status_tag_service.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+class StatusTagService < ::WasteCarriersEngine::BaseService
+  def run(resource:)
+    return [] if resource.blank?
+
+    @resource = resource
+
+    tags = []
+    tags << metadata_status
+    tags << pending_conviction_check
+    tags << pending_payment
+    tags << stuck
+
+    tags.compact
+  end
+
+  private
+
+  def metadata_status
+    return reg_metadata_status if @resource.is_a?(WasteCarriersEngine::Registration)
+    return transient_reg_metadata_status if @resource.is_a?(WasteCarriersEngine::TransientRegistration)
+  end
+
+  def reg_metadata_status
+    return :in_progress if @resource.metaData.PENDING?
+
+    @resource.metaData.status.downcase.to_sym
+  end
+
+  def transient_reg_metadata_status
+    return :revoked if @resource.metaData.REVOKED?
+    return :refused if @resource.metaData.REFUSED?
+    return :in_progress unless submitted_renewal?
```

Comment:
I think we can get away with just calling `renewal_application_submitted?` as we already know it's a transient registration here

```suggestion
    return :in_progress unless @resource.renewal_application_submitted?
```